hs_core/views/resource_folder_hierarchy.py

```diff
@@ -341,6 +347,82 @@ def data_store_folder_unzip_public(request, pk, pathname):
     return data_store_folder_unzip(request, res_id=pk, zip_with_rel_path=sys_pathname)
 
 
+@api_view(['POST'])
+def data_store_add_reference(request):
+    """
+    create the reference url file, add the url file to resource, and add the url to
+    metadata accordingly for easy later retrieval
+    :param request:
+    :return: JsonResponse with status code and message
+    """
+
+    res_id = request.POST.get('res_id', None)
+    curr_path = request.POST.get('curr_path', None)
+    ref_name = request.POST.get('ref_name', None)
```

Comment:
I think these parameters (ref_name and ref_url) should be consistent for both add and edit.

Reply:
Also, any reason we can't have a default curr_path at the root of the resource?

Reply:
@sblack-usu Good point. I'll make these parameters consistent for both add and edit. No particular reason we can't have a default curr_path, but I prefer to have the path being passed in explicitly as what file browser on the UI does. However, if you have a reason to have a default curr_path at the root of the resource and make this parameter optional, I can make this change. I feel it is good to have users to pass in this curr_path parameter explicitly to avoid potential mistake of forgetting to pass in a path.

Reply:
@sblack-usu I took back with my previous comment on making these parameters consistent for both add and edit after looking into the code. For the add action, ```ref_name``` is the name that identifies the URL and will be used as the base file name with appended ```.url``` extension. However, for the edit action, on the UI, only a new referenced url text box is presented to users for editing the url. The ```url_filename``` is implicit and passed in by file browser with whatever .url file being selected for editing. The edit referenced url menu only shows up when users right click on a .url file. So ```url_filename``` in the edit action reflects what it is really, which is different from ```ref_name``` in the context of the add action. So I think it is OK to keep the names different between the add and edit actions since they are different between the two actions.

Reply:
ok